### ISSUE TITLE
Windows: Use ConPty from OpenConsole instead of ConHost

### ISSUE THREAD

**Issue by levirs565** [alacritty/alacritty#4794]
I have some issue with Alacritty. Then, I try reproduce the issue with WezTerm and Windows Terminal. However, I cannot reproduce the issue in Wezterm and Windows Terminal. Even, I make WezTerm config equally with Alacritty. However, I am still cannot repoduce the issue. Then, It turns out that WezTerm and Windows Terminal use ConPty from OpenConsole whereas Alacritty use ConPty from ConHost.

OpenConsole is ConHost clone. Its maintained by Microsoft. OpenConsole source code is come from ConHost source code. OpenConsole have more feature than ConHost. You can see difference in https://github.com/microsoft/terminal/issues/8101. 

You can see how WezTerm use OpenConsole in this file https://github.com/wez/wezterm/blob/a034730373c341aa078765c081fcb32168b1974f/pty/src/win/psuedocon.rs .

My issue is gitui not rendered correctly **only when `$env:TERM`  is not blank**.  

![Annotation 2021-02-15 153022](https://user-images.githubusercontent.com/42236775/107922551-bfe87600-6fa2-11eb-8ad1-38d130e62a48.png)

When I set `$env:TERM` blank. It works.

![Annotation 2021-02-15 153022](https://user-images.githubusercontent.com/42236775/107923144-a693f980-6fa3-11eb-93a4-6919c06890c1.png)

This issue can be reproduced using PowerShell Core directly (Directly ConHost). 

![Annotation 2021-02-15 153556](https://user-images.githubusercontent.com/42236775/107923070-8d8b4880-6fa3-11eb-9a04-98f930124cb8.png)

When `$env:TERM` in PS Core is blank. Gitui work. This issue cannot reproduced in Windows Terminal and WezTerm even when `$env:TERM` is `xterm-256color`.

![Annotation 2021-02-15 154945](https://user-images.githubusercontent.com/42236775/107924394-72213d00-6fa5-11eb-8c31-4e8c439c2c38.png)

Then, I create experiment. I rename `conpty.dll` to `conpty_12.dll` in WezTerm installation. This will make WezTerm use ConHost. Yeah, WezTerm not render Gitui correclty when `$env:TERM` is `xterm-256color`.

![Annotation 2021-02-15 155220](https://user-images.githubusercontent.com/42236775/107924623-dc39e200-6fa5-11eb-86cf-91c56321c0e6.png)

Then I renamed back `conpty_12.dll` to `conpty.dll`. Gitui work again as before in WezTerm.

Then I try run PS Core direclty from OpenConsole and compare what happenned in ConHost. I see OpenConsole and ConHost send differrent text when running Gitui with `$env:TERM=xterm-256color`. This is text that sent by OpenConsole and ConHost.

```
ConHost
[?1049h[?25l[2J[1;2H[38;5;8m [4m[38;5;15mStatus [1][24m[38;5;8m  |  Log [2]  |  Stashing [3]  |  Stashes [4]                                                              [2;2H──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────[3;1H[39m┌[1mUnstaged Changes [w][22m──────────────────────────────────────┐[38;5;8m┌Diff: ────────────────────────────────────────────────────┐[4;1H[39m│[38;5;8mLoading ...[4;60H[39m│[38;5;8m│[4;120H│[5;1H[39m│[5;60H│[38;5;8m│[5;120H│[6;1H[39m│[6;60H│[38;5;8m│[6;120H│[7;1H[39m│[7;60H│[38;5;8m│[7;120H│[8;1H[39m│[8;60H│[38;5;8m│[8;120H│[9;1H[39m│[9;60H│[38;5;8m│[9;120H│[10;1H[39m│[10;60H│[38;5;8m│[10;120H│[11;1H[39m│[11;60H│[38;5;8m│[11;120H│[12;1H[39m│[12;60H│[38;5;8m│[12;120H│[13;1H[39m│[13;60H│[38;5;8m│[13;120H│[14;1H[39m│[14;60H│[38;5;8m│[14;120H│[15;1H[39m│[15;60H│[38;5;8m│[15;120H│[16;1H[39m│[16;60H│[38;5;8m│[16;120H│[17;1H[39m│[17;60H│[38;5;8m│[17;120H│[18;1H[39m└──────────────────────────────────────────────↑2[18;51H↓0[18;54H{main}┘[38;5;8m│[18;120H│[19;1H┌Staged Changes [s]────────────────────────────────────────┐│[19;120H│[20;1H│Loading ...[20;60H││[20;120H│[21;1H│[21;60H││[21;120H│[22;1H│[22;60H││[22;120H│[23;1H│[23;60H││[23;120H│[24;1H│[24;60H││[24;120H│[25;1H│[25;60H││[25;120H│[26;1H│[26;60H││[26;120H│[27;1H│[27;60H││[27;120H│[28;1H│[28;60H││[28;120H│[29;1H└──────────────────────────────────────────────────────────┘└──────────────────────────────────────────────────────────┘[30;1H[48;5;4mStage All [a][30;15HStage Item [⏎][30;30HReset Item [⇧D][30;46HIgnore [i][30;57H[38;5;15mBranches [b][30;70HPush [p][30;79H[38;5;8mEdit Item [e][30;93HDiff [→][30;102HNav [↑↓→←][30;113H[39m[49mmore[30;118H[.][39m[49m[0m[?25l[1;1H [39m[49m[0m[39m[49m[0m[?25l[1;1H [39m[49m[0m[4;2H[38;5;15m[48;5;4m  ▾alacritty                                              [5;2H[38;5;3m[49mM   alacritty.yml[20;2H[39m           [30;1H[38;5;15m[48;5;4mStage All [a][30;15HStage Item [⏎][30;30HReset Item [⇧D][30;46HIgnore [i][30;102HNav [↑↓→←][39m[49m[0m[?25l[1;1H [39m[49m[0m[39m[49m[0m[?25l[1;1H [39m[49m[0m[1;1H [39m[49m[0m[1;1H [39m[49m[0m[1;1H [39m[49m[0m[1;1H [39m[49m[0m[1;1H [39m[49m[0m[1;1H [39m[49m[0m[1;1H [39m[49m[0m[1;1H [39m[49m[0m[1;1H [39m[49m[0m[1;1H [39m[49m[0m[1;1H [39m[49m[0m[1;1H [39m[49m[0m[1;1H [39m[49m[0m[1;1H [39m[49m[0m[1;1H [39m[49m[0m[1;1H [39m[49m[0m[1;1H [39m[49m[0m[1;1H [39m[49m[0m[1;1H [39m[49m[0m[39m[49m[0m[?25l[1;1H [39m[49m[0m[?25h[?1049l
OpenConsole
←[?1049h←[?25l←[2J←[1;2H←[38;5;8m ←[4m←[38;5;15mStatus [1]←[24m←[38;5;8m  |  Log [2]  |  Stashing [3]  |  Stashes [4]                                                              ←[2;2H──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────←[3;1H←[39m┌←[1mUnstaged Changes [w]←[22m──────────────────────────────────────┐←[38;5;8m┌Diff: ────────────────────────────────────────────────────┐←[4;1H←[39m│←[38;5;8mLoading ...←[4;60H←[39m│←[38;5;8m│←[4;120H│←[5;1H←[39m│←[5;60H│←[38;5;8m│←[5;120H│←[6;1H←[39m│←[6;60H│←[38;5;8m│←[6;120H│←[7;1H←[39m│←[7;60H│←[38;5;8m│←[7;120H│←[8;1H←[39m│←[8;60H│←[38;5;8m│←[8;120H│←[9;1H←[39m│←[9;60H│←[38;5;8m│←[9;120H│←[10;1H←[39m│←[10;60H│←[38;5;8m│←[10;120H│←[11;1H←[39m│←[11;60H│←[38;5;8m│←[11;120H│←[12;1H←[39m│←[12;60H│←[38;5;8m│←[12;120H│←[13;1H←[39m│←[13;60H│←[38;5;8m│←[13;120H│←[14;1H←[39m│←[14;60H│←[38;5;8m│←[14;120H│←[15;1H←[39m│←[15;60H│←[38;5;8m│←[15;120H│←[16;1H←[39m│←[16;60H│←[38;5;8m│←[16;120H│←[17;1H←[39m│←[17;60H│←[38;5;8m│←[17;120H│←[18;1H←[39m└──────────────────────────────────────────────↑2←[18;51H↓0←[18;54H{main}┘←[38;5;8m│←[18;120H│←[19;1H┌Staged Changes [s]────────────────────────────────────────┐│←[19;120H│←[20;1H│Loading ...←[20;60H││←[20;120H│←[21;1H│←[21;60H││←[21;120H│←[22;1H│←[22;60H││←[22;120H│←[23;1H│←[23;60H││←[23;120H│←[24;1H│←[24;60H││←[24;120H│←[25;1H│←[25;60H││←[25;120H│←[26;1H│←[26;60H││←[26;120H│←[27;1H│←[27;60H││←[27;120H│←[28;1H│←[28;60H││←[28;120H│←[29;1H└──────────────────────────────────────────────────────────┘└──────────────────────────────────────────────────────────┘←[30;1H←[48;5;4mStage All [a]←[30;15HStage Item [⏎]←[30;30HReset Item [⇧D]←[30;46HIgnore [i]←[30;57H←[38;5;15mBranches [b]←[30;70HPush [p]←[30;79H←[38;5;8mEdit Item [e]←[30;93HDiff [→]←[30;102HNav [↑↓→←]←[30;113H←[39m←[49mmore←[30;118H[.]←[39m←[49m←[0m←[?25l←[1;1H ←[39m←[49m←[0m←[39m←[49m←[0m←[?25l←[1;1H ←[39m←[49m←[0m←[4;2H←[38;5;15m←[48;5;4m  ▾alacritty                                              ←[5;2H←[38;5;3m←[49mM   alacritty.yml←[20;2H←[39m           ←[30;1H←[38;5;15m←[48;5;4mStage All [a]←[30;15HStage Item [⏎]←[30;30HReset Item [⇧D]←[30;46HIgnore [i]←[30;102HNav [↑↓→←]←[39m←[49m←[0m←[?25l←[1;1H ←[39m←[49m←[0m←[39m←[49m←[0m←[?25l←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[1;1H ←[39m←[49m←[0m←[39m←[49m←[0m←[?25l←[1;1H ←[39m←[49m←[0m←[?25h←[?1049l

```

### System

OS: Windows 10 1909
Version: alacritty 0.7.1 (a852eb8)

**Comment by chrisduerr** [comment]
Alacritty uses Windows' official API, if that doesn't work, you can send that complaint to Microsoft.

**Comment by levirs565** [comment]
OpenConsole is also created and maintained by Microsoft. OpenConsole bleeding edge version of Console Host. Change made in OpenConsole is merged to ConsoleHost after couple of month. **OpenConsole also have same API with ConsoleHost**. This is OpenConsole source code https://github.com/microsoft/terminal/tree/main/src/ . This make old Windows user can use ConPty.

**Comment by levirs565** [comment]
At least you can add the option to use ConPty from the Open Console.

**Comment by fredizzimo** [comment]
@levirs565, I added support for that and sent a pull request last autumn, but it was immediately closed. https://github.com/alacritty/alacritty/pull/4501. And before that I manually patched conpty, but I don't think that is possible anymore.  

I have been running with that setup now for over a year without any issues I can contribute to that. Without it I could not run Alacritty at all, since it was unusuable for me.
